compiler/rustc_resolve/src/def_collector.rs

```diff
@@ -191,7 +221,9 @@ impl<'a, 'b> visit::Visitor<'a> for DefCollector<'a, 'b> {
         };
         self.create_def(param.id, def_path_data, param.ident.span);
 
-        visit::walk_generic_param(self, param);
+        self.with_impl_trait(ImplTraitContext::Universal(self.parent_def), |this| {
+            visit::walk_generic_param(this, param)
+        });
```

Comment:
Could you give an example of this case?
It's clear that function parameters enable universal impl-trait, but generic parameters? Where's the analogous code in AST lowering?

Reply:
Added a comment.